test/app-luatest/gh_6817_console_memory_leak_test.lua

```diff
@@ -0,0 +1,29 @@
+local console = require('console')
+local ffi = require('ffi')
+local fiber = require('fiber')
+local t = require('luatest')
+local g = t.group()
+
+ffi.cdef([[
+    int pipe(int pipefd[2]);
+    int dup2(int oldfd, int newfd);
+    ssize_t write(int fd, const void* buf, size_t count);
+]])
+
+-- Checks that ASAN doesn't detect any memory leaks when console is used.
+g.test_console_mem_leak = function()
+    local fd = ffi.new([[ int[2] ]])
+    t.assert_equals(ffi.C.pipe(fd), 0)
+    t.assert_equals(ffi.C.dup2(fd[0], 0), 0)
```

Comment:
Shouldn't we return the real stdin in, say, after_all() callback?

Reply:
luatest shouldn't use stdin so I think it's okay to close it after the test (don't want to complicate things with dup). Done.

Reply:
Okay.

Ideally we should make this test isolated (when will implement https://github.com/tarantool/luatest/issues/191).